libraries/chain/dbs_budget.cpp

```diff
@@ -0,0 +1,203 @@
+#include <scorum/chain/dbs_budget.hpp>
+#include <scorum/chain/database.hpp>
+#include <scorum/chain/dbs_account.hpp>
+#include <scorum/chain/account_object.hpp>
+
+#include <scorum/chain/budget_objects.hpp>
+#include <scorum/chain/database.hpp>
+
+#include <tuple>
+
+namespace scorum {
+namespace chain {
+
+dbs_budget::dbs_budget(database& db)
+    : _base_type(db)
+{
+    db_impl().get_index<budget_index>().indicies().get<by_owner_name>();
+}
+
+dbs_budget::budget_ids_type dbs_budget::get_budgets(const account_name_type& owner) const
+{
+    budget_ids_type ret;
+
+    auto it_pair = db_impl().get_index<budget_index>().indicies().get<by_owner_name>().equal_range(owner);
+    auto it = it_pair.first;
+    auto it_end = it_pair.second;
+    FC_ASSERT(it != it_end, "budget not found");
+    while (it != it_end)
+    {
+        ret.push_back(it->id);
+        it++;
+    }
+
+    return ret;
+}
+
+const budget_object& dbs_budget::get_budget(budget_id_type id) const
+{
+    try
+    {
+        return db_impl().get<budget_object, by_id>(id);
+    }
+    FC_CAPTURE_AND_RETHROW() // to write __FILE__, __LINE__ from here
+}
+
+const budget_object& dbs_budget::get_any_budget(const account_name_type& owner) const
+{
+    auto it_pair = db_impl().get_index<budget_index>().indicies().get<by_owner_name>().equal_range(owner);
+    auto it_begin = it_pair.first;
+    auto it_end = it_pair.second;
+    FC_ASSERT(it_begin != it_end, "budget not found");
+
+    return *it_begin;
+}
+
+const budget_object& dbs_budget::create_fund_budget(const asset& balance_in_scorum,
+                                                    const share_type& per_block,
+                                                    const time_point_sec& deadline)
+{
+    FC_ASSERT(balance_in_scorum.symbol == SCORUM_SYMBOL, "invalid asset type (symbol)");
```

Comment:
you could create private method and share it between  create_fund_object and create_fund

Reply:
This code
+ FC_ASSERT(balance_in_scorum.symbol == SCORUM_SYMBOL, "invalid asset type (symbol)");
+    FC_ASSERT(balance_in_scorum.amount > 0, "invalid balance_in_scorum");
 +    FC_ASSERT(per_block > 0, "invalid per_block");
 +
 +    auto head_block_num = db_impl().head_block_num();
 +
 +    const budget_object& new_budget = db_impl().create<budget_object>([&](budget_object& budget) {
 +        budget.owner = SCORUM_ROOT_POST_PARENT;
 +        budget.balance = balance_in_scorum;
 +        budget.per_block = per_block;
 +        // allocate cash only after next block generation
 +        budget.last_allocated_block = head_block_num;
 +    });

Reply:
I see no reason to make this private only for 3 asserts. And it will be useful to get assert in place 
(log show exact create_)

Reply:
Maybe it is logically to move:
    FC_ASSERT(balance_in_scorum.amount > 0, "invalid balance_in_scorum");
    FC_ASSERT(per_block > 0, "invalid per_block");
to evaluator

Reply:
Not for asserts, for budget creation. look at my comment higher.

Reply:
I don't agree with you.